assets/src/edit-story/components/library/panes/media/mediaEditDialog.js

```diff
@@ -0,0 +1,202 @@
+  justify-content: left;
+  align-items: flex-start;
+`;
+
+const MetadataTextContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const MediaTitleText = styled.div`
+  font-weight: bold;
+  color: #3c4043;
+`;
+
+const MediaSizeText = styled.div`
+  font-size: 16px;
+  color: #3c4043;
+`;
+
+const Input = styled.input`
+  background: #ffffff;
+  border: 1px solid #d3d4d4;
+  box-sizing: border-box;
+  border-radius: 4px;
+  font-size: 15px;
+  padding: 7px 10px;
+  margin-top: 20px;
+  margin-bottom: 4px;
+`;
+
+const DialogDescription = styled.div`
+  font-size: 12px;
+  line-height: 16px;
+  color: #202124;
+  opacity: 0.54;
+`;
+
+const imageDialogTitle = __('Edit Image', 'web-stories');
+const videoDialogTitle = __('Edit Video', 'web-stories');
+const imageDialogDescription = __(
+  'Describe the appearance and function of an image. Leave empty if the image is purely decorative',
+  'web-stories'
+);
+const videoDialogDescription = __(
+  'Describe the appearance and function of a video. Leave empty if the video is purely decorative',
+  'web-stories'
+);
+
+/**
+ * Displays a dialog where the user can edit the selected media element.
+ *
+ * @param {Object} props Component props.
+ * @param {Object} props.resource Selected media element's resource object.
+ * @param {Object} props.showEditDialog If dialog should be displayed.
+ * @param {Object} props.setShowEditDialog Callback to toggle dialog display.
+ * @return {null|*} The dialog element.
+ */
+function MediaEditDialog({ resource, showEditDialog, setShowEditDialog }) {
+  const {
+    id,
+    src,
+    title,
+    width,
+    height,
+    type,
+    alt,
+    poster,
+    mimeType,
+  } = resource;
+  const {
+    actions: { updateMedia },
+  } = useAPI();
+  const [altText, setAltText] = useState(alt);
+  const [shouldUpdate, setShouldUpdate] = useState(false);
+
+  const handleAltTextChange = useCallback((evt) => {
+    setAltText(evt.target.value);
+  }, []);
+
+  useEffect(() => {
+    const updateMediaItem = async () => {
+      try {
+        await updateMedia(id, { alt_text: altText });
+        setShowEditDialog(false);
+      } catch (err) {
+        // TODO Display error message
+      } finally {
+        setShouldUpdate(false);
+      }
+    };
+    shouldUpdate ? updateMediaItem() : null;
+  }, [altText, id, setShowEditDialog, shouldUpdate, updateMedia]);
+
+  return (
+    <Dialog
+      open={showEditDialog}
+      onClose={() => setShowEditDialog(false)}
+      width={'500px'}
+      title={type == 'image' ? imageDialogTitle : videoDialogTitle}
+      actions={
+        <>
+          <Plain onClick={() => setShowEditDialog(false)}>
+            {__('Cancel', 'web-stories')}
+          </Plain>
+          <Plain onClick={() => setShouldUpdate(true)}>
+            {__('Save', 'web-stories')}
+          </Plain>
+        </>
+      }
+    >
+      <DialogBody>
+        {type == 'image' ? (
+          <Image src={src} alt={alt} loading={'lazy'} />
```

Comment:
Which size image is being used here?

Reply:
Added a small util (`useSmallestImage`) for this!